RULES_DESCRIPTIONS.md

```diff
@@ -192,6 +193,25 @@ _Description_: Packages exposing functions that can stop program execution by ex
 
 _Configuration_: N/A
 
+## defer
+
+_Description_: This rule warns on some common mistakes when using `defer` statement. It currently alerts on the following situations:
```

Comment:
To make them configurable and allow folks to disable some of these checks, maybe it makes sense to have 4 separate rules?